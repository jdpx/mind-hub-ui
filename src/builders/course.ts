import faker from 'faker'
import { Course as CourseType } from 'types/course'

class Course {
    id: string
    title: string

    constructor() {
        this.id = '1'
        this.title = faker.lorem.words(2)
    }

    Build = (): CourseType => ({
        id: this.id,
        title: this.title,
    })
}

export function CourseBuilder() {
    return new Course()
}
