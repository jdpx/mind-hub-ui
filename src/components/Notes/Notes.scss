@import "../../styles/colours.scss";

.notes {
    display: flex;
    flex-direction: row;
    align-items: stretch;
    justify-content: stretch;
    padding: 0 27px 27px;

    textarea {
        flex: 1;
        font-size: 15px;
        line-height: 17px;
        border: 0;
        min-height: 180px;
        padding: 12px;

        background-color: $background-grey;
        border-radius: 5px;

        overflow: auto;
    }
}
