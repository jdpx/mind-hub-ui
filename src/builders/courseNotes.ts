import faker from 'faker'
import { CourseNote as NoteType } from 'types/course'

class CourseNote {
    id: string
    value: string

    constructor() {
        this.id = '1'
        this.value = faker.lorem.words(2)
    }

    Value = (value: string) => {
        this.value = value
        return this
    }

    Build = (): NoteType => ({
        id: this.id,
        value: this.value,
    })
}

export function CourseNoteBuilder() {
    return new CourseNote()
}
