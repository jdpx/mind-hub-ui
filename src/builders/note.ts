import faker from 'faker'
import { Note as NoteType } from 'types/course'

class Note {
    id: string
    value: string

    constructor() {
        this.id = '1'
        this.value = faker.lorem.words(2)
    }

    Build = (): NoteType => ({
        id: this.id,
        value: this.value,
    })
}

export function NoteBuilder() {
    return new Note()
}
