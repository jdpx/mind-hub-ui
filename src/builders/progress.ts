import faker from 'faker'
import { Progress as ProgressType } from 'types/course'

class Progress {
    started: boolean
    sessionsCompleted: number

    constructor() {
        this.started = true
        this.sessionsCompleted = faker.random.number(2)
    }

    HasStarted(started: boolean) {
        this.started = started

        return this
    }

    SessionsCompleted(count: number) {
        this.sessionsCompleted = count

        return this
    }

    Build = (): ProgressType => ({
        started: this.started,
        sessionsCompleted: this.sessionsCompleted,
    })
}

export function ProgressBuilder() {
    return new Progress()
}
