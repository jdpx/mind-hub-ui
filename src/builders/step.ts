import faker from 'faker'
import { Course, Step as StepType, StepProgress } from 'types/course'
import { Question } from '../constants/steps'
import { StepTypes } from '../types/step'

class Step {
    id: string
    title: string
    description: string
    course?: Course
    steps: Step[]
    type: StepTypes
    progress?: StepProgress

    constructor() {
        this.id = faker.lorem.slug()
        this.title = faker.lorem.words(2)
        this.description = faker.lorem.sentences(2)
        this.type = Question
        this.steps = []
    }

    ID = (id: string) => {
        this.id = id

        return this
    }

    WithType(type: StepTypes) {
        this.type = type

        return this
    }

    WithTitle(title: string) {
        this.title = title
        return this
    }

    WithProgress(progress: StepProgress) {
        this.progress = progress

        return this
    }

    Build = (): StepType => ({
        id: this.id,
        title: this.title,
        description: this.description,
        type: this.type,
        progress: this.progress,
    })
}

export function StepBuilder() {
    return new Step()
}
