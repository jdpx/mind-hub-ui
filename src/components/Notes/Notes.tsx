import React from 'react'
import classNames from 'classnames'

import { CourseNote, StepNote } from '../../types/course'
import Section from '../Section/Section'

import './Notes.scss'

interface Props {
    title?: string
    placeholder?: string
    note?: CourseNote | StepNote
    className?: string
    testid?: string
    handleSave?: (val: string) => void
}

export default function Notes({
    title = 'Notes',
    placeholder = 'Enter notes here',
    note,
    handleSave,
    className,
    testid,
}: Props) {
    const onTextAreaBlue = (event: React.FocusEvent<HTMLTextAreaElement>) => {
        const text = event.target.value

        if (text === note?.value) {
            return
        }

        if (handleSave) {
            handleSave(text)
        }
    }

    const noteClass = classNames('notes', {
        [className || '']: !!className,
    })

    return (
        <Section title={title}>
            <div className={noteClass}>
                <textarea
                    placeholder={placeholder}
                    defaultValue={note && note.value}
                    onBlur={onTextAreaBlue}
                    data-testid={testid}
                />
            </div>
        </Section>
    )
}
