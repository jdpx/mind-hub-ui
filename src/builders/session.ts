import faker from 'faker'
import { Course } from 'types/course'
import { Step } from 'types/step'
import { Session as SesssionType } from 'types/session'

class Session {
    private id: string
    private title: string
    private description: string
    private course?: Course
    private steps: Step[]

    constructor() {
        this.id = '2'
        this.title = faker.lorem.words(2)
        this.description = faker.lorem.sentences(2)
        this.steps = []
    }

    WithID = (id: string) => {
        this.id = id

        return this
    }
    WithRandomID = () => {
        this.id = faker.lorem.slug()

        return this
    }

    WithCourse = (course: Course) => {
        this.course = course

        return this
    }

    WithStep = (step: Step) => {
        this.steps.push(step)

        return this
    }

    WithSteps = (steps: Step[]) => {
        this.steps = [...this.steps, ...steps]

        return this
    }

    Build = (): SesssionType => ({
        id: this.id,
        title: this.title,
        description: this.description,
        course: this.course,
        steps: this.steps,
    })
}

export function SessionBuilder() {
    return new Session()
}
