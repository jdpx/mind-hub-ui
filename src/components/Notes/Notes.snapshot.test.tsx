import React from 'react'
import { render } from '@testing-library/react'
import Notes from './Notes'
import { CourseNoteBuilder } from '../../builders/courseNotes'

describe('Notes Snapshots', () => {
    const props = {
        title: 'My notes',
        placeholder: 'Helloworld',
        testid: 'my-notes',
    }

    it('renders default note component', () => {
        const { asFragment } = render(<Notes {...props} />)

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders note with custom class', () => {
        const { asFragment } = render(<Notes className="custom class" {...props} />)

        expect(asFragment()).toMatchSnapshot()
    })

    describe('given custom note', () => {
        it('renders note text', () => {
            const note = CourseNoteBuilder().Value('My notes value').Build()

            const { asFragment } = render(<Notes note={note} {...props} />)

            expect(asFragment()).toMatchSnapshot()
        })
    })
})
