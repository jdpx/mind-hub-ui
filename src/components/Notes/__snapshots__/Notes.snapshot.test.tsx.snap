// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes Snapshots given custom note renders note text 1`] = `
<DocumentFragment>
  <div
    class="content-wrapper withTitle"
  >
    <h3
      class="section-title"
    >
      My notes
    </h3>
    <div
      class="notes"
    >
      <textarea
        data-testid="my-notes"
        placeholder="Helloworld"
      >
        My notes value
      </textarea>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Notes Snapshots renders default note component 1`] = `
<DocumentFragment>
  <div
    class="content-wrapper withTitle"
  >
    <h3
      class="section-title"
    >
      My notes
    </h3>
    <div
      class="notes"
    >
      <textarea
        data-testid="my-notes"
        placeholder="Helloworld"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Notes Snapshots renders note with custom class 1`] = `
<DocumentFragment>
  <div
    class="content-wrapper withTitle"
  >
    <h3
      class="section-title"
    >
      My notes
    </h3>
    <div
      class="notes custom class"
    >
      <textarea
        data-testid="my-notes"
        placeholder="Helloworld"
      />
    </div>
  </div>
</DocumentFragment>
`;
